File: Documentation/5.3/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Revisions/ClientAPI/Operations/GetRevisions.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Operations.Revisions;
using Raven.Documentation.Samples.Orders;

namespace Raven.Documentation.Samples.DocumentExtensions.Revisions.ClientAPI.Operations
{
    public class GetRevisions
    {
        public GetRevisions()
        {
            using (var documentStore = new DocumentStore())
            {
                #region getAllRevisions
                // Define the get revisions operation, pass the document id
                var getRevisionsOp = new GetRevisionsOperation<Company>("Companies/1-A");
                
                // Execute the operation by passing it to Operations.Send
                RevisionsResult<Company> revisions = documentStore.Operations.Send(getRevisionsOp);

                // The revisions info:
                List<Company> allRevisions = revisions.Results; // All the revisions
                int revisionsCount = revisions.TotalResults;    // Total number of revisions
                #endregion
            }
            
            using (var documentStore = new DocumentStore())
            {
                #region getRevisionsWithPaging
                var start = 0;
                var pageSize = 100;
                
                while (true)
                {
                    // Execute the get revisions operation
                    // Pass the document id, start & page size to get
                    RevisionsResult<Company> revisions = documentStore.Operations.Send(
                        new GetRevisionsOperation<Company>("Companies/1-A", start, pageSize));
                    
                    {
                        // Process the retrieved revisions here
                    }
                    
                    if (revisions.Results.Count < pageSize)
                        break; // No more revisions to retrieve

                    // Increment 'start' by page-size, to get the "next page" in next iteration
                    start += pageSize;
                }
                #endregion
            }
            
            using (var documentStore = new DocumentStore())
            {
                #region getRevisionsWithPagingParams
                var parameters = new GetRevisionsOperation<Company>.Parameters
                {
                    Id = "Companies/1-A",
                    Start = 0,
                    PageSize = 100
                };

                RevisionsResult<Company> revisions = documentStore.Operations.Send(
                    new GetRevisionsOperation<Company>(parameters));
                #endregion
            }
        }
        
        public async Task GetRevisionsAsync()
        {
            using (var documentStore = new DocumentStore())
            {
                #region getAllRevisions_async
                // Define the get revisions operation, pass the document id
                var getRevisionsOp = new GetRevisionsOperation<Company>("Companies/1-A");
                
                // Execute the operation by passing it to Operations.Send
                RevisionsResult<Company> revisions = await documentStore.Operations.SendAsync(getRevisionsOp);

                // The revisions info:
                List<Company> allRevisions = revisions.Results; // All the revisions
                int revisionsCount = revisions.TotalResults;    // Number of revisions
                #endregion
            }

            using (var documentStore = new DocumentStore())
            {
                #region getRevisionsWithPaging_async
                var start = 0;
                var pageSize = 100;
                
                while (true)
                {
                    // Execute the get revisions operation
                    // Pass the document id, start & page size to get
                    RevisionsResult<Company> revisions = await documentStore.Operations.SendAsync(
                        new GetRevisionsOperation<Company>("Companies/1-A", start, pageSize));
                    {
                        // Process the retrieved revisions here
                    }
                    
                    if (revisions.Results.Count < pageSize)
                        break; // No more revisions to retrieve

                    // Increment 'start' by page-size, to get the "next page" in next iteration
                    start += pageSize;
                }
                #endregion
            }

            using (var documentStore = new DocumentStore())
            {
                #region getRevisionsWithPagingParams_async
                var parameters = new GetRevisionsOperation<Company>.Parameters
                {
                    Id = "Companies/1-A",
                    Start = 0,
                    PageSize = 100
                };

                RevisionsResult<Company> revisions = await documentStore.Operations.SendAsync(
                    new GetRevisionsOperation<Company>(parameters));
                #endregion
            }
        }
    }
}

File: Documentation/5.4/Samples/nodejs/documentExtensions/revisions/client-api/operations/getRevisions.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();

async function getRevisions() {
    {
        //region getAllRevisions
        // Define the get revisions operation, pass the document id
        const getRevisionsOp = new GetRevisionsOperation("Companies/1-A");

        // Execute the operation by passing it to operations.send
        const revisions = await documentStore.operations.send(getRevisionsOp);

        // The revisions info:
        const allRevisions = revisions.results;        // All the revisions
        const revisionsCount = revisions.totalResults; // Total number of revisions
        //endregion
    }
    {
        //region getRevisionsWithPaging
        const parameters = {
            start: 0,
            pageSize: 100
        };
        
        while (true)
        {
            // Execute the get revisions operation
            // Pass parameters with document id, start & page size
            const revisions = await documentStore.operations.send(
                new GetRevisionsOperation("Companies/1-A", parameters));
            
            {
                // Process the retrieved revisions here
            }

            if (revisions.results.length < parameters.pageSize)
                break; // No more revisions to retrieve

            // Increment 'start' by page-size, to get the "next page" in next iteration
            parameters.start += parameters.pageSize;
        }
        //endregion
    }
}

{
    //region syntax_1  
    // Available overloads:
    const getRevisionsOp = new GetRevisionsOperation(id);
    const getRevisionsOp = new GetRevisionsOperation(id, parameters);
    //endregion

    //region syntax_2
    // The parameters object
    {
        start,   // Revision number to start from
        pageSize // Number of revisions to get
    }
    //endregion

    //region syntax_3
    class RevisionsResult
    {
        results;      // The retrieved revisions
        totalResults; // Total number of revisions that exist for the document
    }
    //endregion
}
